server/backends/disk_cache/disk_cache.go

```diff
@@ -64,13 +105,46 @@ func (c *DiskCache) WithPrefix(prefix string) interfaces.Cache {
 	}
 
 	return &DiskCache{
-		prefix:    newPrefix,
-		partition: c.partition,
+		env:        c.env,
+		prefix:     newPrefix,
+		partition:  c.partition,
+		partitions: c.partitions,
 	}
 }
 
+func (c *DiskCache) WithRemoteInstanceName(ctx context.Context, remoteInstanceName string) interfaces.Cache {
```

Comment:
Since you're already introducing a new method -- it seems like a good time to change the interface to return an error here if the partition is invalid or unrecognized

Reply:
Good point, done.